q2_fmt/_engraftment.py

```diff
@@ -460,3 +419,63 @@ def _independent_dists(diversity_measure, metadata,
         nominal_df = nominal_df[['id', 'measure', 'group', 'A', 'B']]
 
     return nominal_df
+
+
+# Helper Function For to-baseline datafiltering
+def _get_to_baseline_ref(time_col, baseline_timepoint, time_column,
+                         subject_column, metadata):
+
+    temp_baseline_ref = []
+    reference_list = []
+    baseline_ref_df = pd.DataFrame()
+    # All valid FMT samples have to have a time column
+    metadata = metadata.to_dataframe()[~time_col.isna()]
+    if float(baseline_timepoint) not in metadata[time_column].values:
+        raise AssertionError("The provided baseline timepoint"
+                             f" {baseline_timepoint} was not"
+                             f" found in `metadata` "
+                             f" column {time_column}.")
+    for sub, samples in metadata.groupby([subject_column]):
+        reference = \
+            samples[samples[
+                time_column] == float(baseline_timepoint)].index.to_list()
+        if len(reference) > 1:
+            raise ValueError("More than one baseline sample was found per"
+                             " subject. Only one baseline sample can be"
+                             " used as a reference. Please group baseline"
+                             " replicates.")
+        elif len(reference) == 0:
+            # If there is no baseline for a subject,
+            # This will either drop with filter-missing-references or
+            # or error and say that they need to pass
+            # filter-missing-references
+            reference = [np.nan]
+        temp_baseline_ref = temp_baseline_ref + samples.index.to_list()
+        reference_list = \
+            reference_list + (reference * len(samples.index.to_list()))
+    # I dont see any way that this hits because of my above assertion but
+    # I think its a good check so I am leavig it in.
+    if len(reference_list) == 0:
+        raise AssertionError("No baseline samples"
+                             " were found in the metadata."
+                             " Please confirm that a valid"
+                             " baseline timepoint was given.")
+    if pd.Series(reference_list).isnull().all():
+        raise AssertionError("No baseline samples"
+                             " were connected via subject."
+                             " Please confirm that all valid"
```

Comment:
This sentence was a bit confusing - I'm assuming this should say something like 'confirm that all baseline timepoints have samples connected with a subject' or something like that?